docs/se/guides/07_db_client.adoc

```diff
@@ -0,0 +1,563 @@
+    delete-book: "DELETE FROM LIBRARY WHERE NAME = ?"
+----
+<1> Source property support two values: jdbc and mongo.
+<2> Connection detail we used to set up H2.
+<3> SQL statements to manage the database.
+<4> The ping statement is mandatory because of health check.
+
+=== Build and set up Helidon DB Client
+
+In the application Main.class, an instance of DbClient is created based on the configuration from
+application.yaml.
+
+[source,java]
+.Create a DbClient in the Main.startServer method:
+----
+import io.helidon.dbclient.metrics.DbClientMetrics; // <1>
+import io.helidon.dbclient.DbClient;
+
+...
+
+Config config = Config.create(); // Landmark to add DB client
+
+DbClient dbClient = DbClient.builder()
+                .config(config.get("db"))       // <2>
+                .addService(DbClientMetrics.counter().statementNames("ping"))   // <3>
+                .build();
+----
+<1> Add import statements
+<2> Configure the DB Client with the "db" section of application.yaml.
+<3> Add a counter for metrics.
+
+The DB Client metric counter will be executed only for ping statement, and check how many times it was invoked.
+At this point, the database is empty, and needs to be initialised. To achieve that, the DB Client can be used
+to create a table in the database.
+
+[source,java]
+.Insert a createTable method below the dbClient:
+----
+DbClient dbClient = DbClient.builder()
+                .config(config.get("db"))
+                .addService(DbClientMetrics.counter().statementNames("ping"))
+                .build();
+
+createTable(dbClient);
+----
+
+[source,java]
+.Use the DbClient to build a table:
+----
+private static void createTable(DbClient dbClient) {
+        dbClient.execute(exec -> exec.namedDml("create-table"))     // <1>
+                .await();
+    }
+----
+<1> Use the "create-table" script to build a table with book name and information.
+
+The createTable is called only once and create an empty table with two columns: name and info. A new service can
+manage request to interact with this table which represent our library.
+The services are registered in the createRouting method.
+
+[source,java]
+.Modify the createRouting method:
+----
+import io.helidon.dbclient.health.DbClientHealthCheck;
+
+...
+
+WebServer server = WebServer.builder(createRouting(config, dbClient))   // <1>
+                .config(config.get("server"))
+                .addMediaSupport(JsonpSupport.create())
+                .build();
+
+...
+
+private static Routing createRouting(Config config, DbClient dbClient) {
+        MetricsSupport metrics = MetricsSupport.create();
+        GreetService greetService = new GreetService(config);
+        LibraryService libraryService = new LibraryService(dbClient); // <2>
+        HealthSupport health = HealthSupport.builder()
+                .addLiveness(HealthChecks.healthChecks())   // Adds a convenient set of checks
+                .addLiveness(DbClientHealthCheck.create(dbClient))  // <3>
```

Comment:
DbClient Health check API will be changed in near future. But it's task for myself to update this doc properly.